frontend/src/components/ConstraintForm/ClassAccordionItem.vue

```diff
@@ -0,0 +1,55 @@
+<script>
+import { AccordionItem, AccordionList } from 'vue3-rich-accordion';
+import SubjectAccordionItem from './SubjectAccordionItem.vue';
+import { useFormStore } from '@/stores/form';
+
+export default {
+  props: {
+    index: { type: Number, default: -1 },
+  },
+  setup(props) {
+    const formStore = useFormStore();
+
+    const accordionId = `${props.index}`;
+
+    return { formStore, accordionId };
+  },
+  components: {
+    AccordionList,
+    AccordionItem,
+    SubjectAccordionItem,
+  },
+};
+</script>
+
+<template>
+  <accordion-item
+    :id="accordionId"
+    :default-opened="accordionId in formStore.accordionState ? formStore.accordionState[accordionId] : true"
+    class="class-accordion"
+  >
+    <template #summary>
+      <input
+        v-model="formStore.form.classes[index].name"
```

Comment:
If `formStore.form.classes[index]` were store in a variable in the `setup` then it would be easier to use in the template here and on line 47.